### PR TITLE
Fix VRMenu flashing

### PR DESCRIPTION
**Pull request by thoys**
Ported `handControllerPointer.js` VR Menu opening at desired position over to C++ to stop it from flashing. 

Added 3 controller actions for different context menu positions:
```js
Actions.ContextMenuCentered // Gamepad start button press
Actions.ContextMenuLeftLaser // Left handcontroller menu button press
Actions.ContextMenuRightLaser // Right handcontroller menu button press
```

## Testplan

This test requires HMD, handcontrollers and an Xbox controller or other supported gamepad

1. Test if the VR menu stopped from flashing.
2. Desktop mode test:
    - Check the gamepad is creating the menu at the cursor position on press of start
    - (optional) If in possession of hydra's try pointing at the 2D screen, press the menu button on the held controller and notice that the menu position depends on your aim direction.
3. HMD mode test:
    - The gamepad should create the menu in the direction that is being looked at press of start button
    - The hand controllers should create the VRMenu in the direction that is pointed at after pressing the controllers menu button.
    - Repeat the actions above a few times in multiple directions, to confirm that the menu positioning algorithm compares with the current release build.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 39ac334c9571a273b9d4398fb164154e8ef21d50 available at: https://gist.github.com/283f1cfd2239c45fef84f30e78258f41<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9324-39ac334.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9324-39ac334.dmg

**Comment by ZappoMan** [comment]
Warnings/errors

On Unix
```
CompositorHelper.cpp:434, GNU C Compiler 4 (gcc), Priority: Normal
implicit conversion from ‘float’ to ‘double’ to match other operand of binary expression [-Wdouble-promotion]
```

On Windows
```
HmdDisplayPlugin.cpp:415, MSBuild, Priority: Normal
'argument' : conversion from 'size_t' to 'int', possible loss of data
```

**Comment by thoys** [comment]
@ZappoMan not sure why it tries to convert from float to double there, I'll try `glm::tan()` , it might pick another `tan` function that is defined in linux. Or am I missing something?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8aac49b39a661e4037be6cc4ac14fac49428a65c available at: https://gist.github.com/283f1cfd2239c45fef84f30e78258f41<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9324-8aac49b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9324-8aac49b.dmg

**Comment by ZappoMan** [comment]
Error on mac

```
In file included from /Users/jenkins/Jenkins/workspace/beta_pr-master/label/osx/libraries/display-plugins/src/display-plugins/hmd/HmdDisplayPlugin.h:20:
/Users/jenkins/Jenkins/workspace/beta_pr-master/label/osx/libraries/display-plugins/src/display-plugins/hmd/../CompositorHelper.h:24:10: fatal error: 'controllers/InputDevice.h' file not found
#include <controllers/InputDevice.h>
         ^
1 error generated.
```

**Comment by thoys** [comment]
@ZappoMan it works for me in OSX, does it skip cmake re-generate? I've added a macro in the display-plugins cmake file `include_hifi_library_headers(controllers)`

**Comment by hifi-gustavo** [comment]
Build console output for commit: 68e6d32b13ccfab8c38e6a91b41ab7d24302f802 available at: https://gist.github.com/283f1cfd2239c45fef84f30e78258f41<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9324-68e6d32.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9324-68e6d32.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 93dcda4c7d3e01ceb5af363a0b987c0fe349e579 available at: https://gist.github.com/283f1cfd2239c45fef84f30e78258f41<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9324-93dcda4.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9324-93dcda4.dmg

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -1061,8 +1061,12 @@ Application::Application(int& argc, char** argv, QElapsedTimer& startupTimer, bo
                 if (!offscreenUi->navigationFocused()) {
                     toggleMenuUnderReticle();
                 }
-            } else if (action == controller::toInt(controller::Action::CONTEXT_MENU)) {
-                toggleMenuUnderReticle();
+            }
```
Comment (ZappoMan) [inline interface/src/Application.cpp]: coding standard: brace placement `} else {`

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6100c2ee9c500e3b753b6f03ecf97c29f3852b24 available at: https://gist.github.com/283f1cfd2239c45fef84f30e78258f41<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9324-6100c2e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9324-6100c2e.dmg

**Comment by OmegaHeron** [comment]
QA Fail

Yes, it fixes the flicker issue, but -- it changes how menu displays completely from method in master.

Master - menu appears in desktop mode where mouse cursor is or in HMD in center of HMD display (lookat point).

This PR - menu will NOT appear at all until you've used a paddle button to bring up a laser beam then menu appears in line with laser.  It does not respect displaying at center of HMD lookat view.

**Comment by OmegaHeron** [comment]
@thoys look at #8976 and (not about xbox controller but doe reference keeping VR menu in center of HMD display) #9123

**Pull request closed without merging**